src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,234 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tale;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        if (size > 1) {
+            Node<T> prevTale = tale;
+            tale = new Node(prevTale, value, null);
+            prevTale.next = tale;
+        } else if (size == 0) {
+            Node<T> fistNode = new Node<>(null, value, null);
+            head = fistNode;
+            tale = fistNode;
+        } else if (size == 1) {
+            Node<T> secondNode = new Node<>(head, value, null);
+            tale = secondNode;
+            head.next = secondNode;
+
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size && size > 1) {
+            throw new IndexOutOfBoundsException("Wrong index!");
+        }
+        if (index == size) {
+            this.add(value);
+            return;
+        } else if (index == 0) {
+            if (size == 0) {
+                this.add(value);
+                return;
+            }
+            Node<T> initialHeadNode = head;
+            head = new Node<>(null, value, initialHeadNode);
+            initialHeadNode.prev = head;
+            size++;
+            return;
+        }
+        if (index < (size - 1) / 2) {
+            int indexCounter = 0;
+            Node<T> tempNode = head.next;
+            while (indexCounter != index) {
+                indexCounter++;
+                if (indexCounter == index) {
+                    Node<T> toAdd = new Node<>(tempNode.prev, value, tempNode);
+                    tempNode.prev.next = toAdd;
+                    tempNode.prev = toAdd;
+                    size++;
+                    return;
+                }
+                tempNode = tempNode.next;
+
+            }
+        } else {
+            int indexCounter = size;
+            Node<T> tempNode = tale;
+            while (indexCounter != index) {
+                indexCounter--;
+                if (indexCounter == index) {
+                    Node<T> toAdd = new Node<>(tempNode.prev, value, tempNode);
+                    tempNode.prev.next = toAdd;
+                    tempNode.prev = toAdd;
+                    size++;
+                    return;
+                }
+                tempNode = tempNode.prev;
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
can be simplified to 
```suggestion
        for (T element : list) {
            add(element);
        }
```